kamu-core/src/infra/utils/simple_transfer_protocol.rs

```diff
@@ -102,12 +127,25 @@ impl SimpleTransferProtocol {
             }
         };
 
+        let old_head = dst_head.clone();
         let num_blocks = blocks.len();
 
+        // Create dataset if necessary using the source Seed block
+        let (dst, dst_head) = if let Some(dst) = maybe_dst {
+            (dst, dst_head)
+        } else {
+            let seed_block = match blocks.pop() {
+                Some((_, seed_block)) => seed_block,
+                _ => unreachable!(),
+            };
```

Comment:
At this point you have no idea whether it is a seed block or not, so the variable should be called somehow differently until the assumption is validated.